### PR TITLE
Spanish translations for 404 and hotlines pages

### LINKED ISSUE #445: localize the 404 page
In #444 , the translations for the home page were moved to our shared package. We're now ready to enable localization to the rest of the web package.

## Translations
https://docs.google.com/document/d/159Lje8l2kHIXIFcLZ0wQJKW2YrYO8mihw3dS4RdoijI/edit?usp=sharing

## Content
https://github.com/CodeForBoulder/upswyng/blob/master/packages/web/src/components/NotFound.tsx

## Checklist
- [ ] translations for both English and Spanish were added to the [shared package](https://github.com/CodeForBoulder/upswyng/tree/master/packages/common/src/i18n)
- [ ] translation keys were substituted for the correct content

### LINKED ISSUE #446: localize hotlines page
In #444 , the translations for the home page were moved to our shared package. We're now ready to enable localization to the rest of the web package.

## Translations

https://docs.google.com/document/d/1UFpzMhhCW-8Xbmmo313edBeoOsHTp7PEmQj1noBnNCQ/edit?usp=sharing

## Content

https://github.com/CodeForBoulder/upswyng/blob/master/packages/web/src/components/Hotlines.tsx

## Checklist
- [ ] translations for both English and Spanish were added to the [shared package](https://github.com/CodeForBoulder/upswyng/tree/master/packages/common/src/i18n)
- [ ] translation keys were substituted for the correct content

### PR DESCRIPTION
**Pull request by paspheeris**
This PR closes #445 and closes #446 

## What does this PR do?
Adds Spanish translations for the 'not found' page and 'hotlines' page.

Definitely let me know if you have any preferences on how the translations are stored or used compared to what I've done, as I'll likely be duplicating this approach for all the other i18n issues!

## How does this PR make you feel? [:link:](http://giphy.com/categories/emotions/)

🇪🇸

### CODE REVIEW

File: packages/common/src/i18n/en.json [hunk 1]
```diff
@@ -11,5 +15,67 @@
...
+    },
+    "nationalChildAbuseHotline": {
+      "name": "National Child Abuse Hotline",
+      "description": "The Childhelp National Child Abuse Hotline 1-800-4-A-CHILD (1-800-422-4453) is dedicated to the prevention of child abuse. Serving the United States, its territories, and Canada, the hotline is staffed 24 hours a day, 7 days a week with professional crisis counselors who, through interpreters, can provide assistance in over 170 languages. The hotline offers crisis intervention, information, literature, and referrals to thousands of emergency, social service, and support resources. All calls are confidential."
+    },
+    "nationalRunawaySafeline": {
+      "name": "National Runaway Safeline",
+      "description": "For Youth and Teens, The National Runaway Safeline (NRS) is here to listen whether you are thinking of running away or already have. Our services are confidential and nonjudgmental. For Parents and Guardians, NRS can offer you support and help connect you to the right resources for your family. Whether you’re having trouble talking to your child or your child has already run, we can help. We will listen and assist you in developing a plan"
```
Comment (jacobvenable) [inline packages/common/src/i18n/en.json:38]:
```suggestion
      "description": "For Youth and Teens, The National Runaway Safeline (NRS) is here to listen whether you are thinking of running away or already have. Our services are confidential and nonjudgmental. For Parents and Guardians, NRS can offer you support and help connect you to the right resources for your family. Whether you’re having trouble talking to your child or your child has already run, we can help. We will listen and assist you in developing a plan."
```
Missing period in the original, not this PR 😅

File: packages/common/src/i18n/es.json [hunk 2]
```diff
@@ -11,5 +15,67 @@
...
+    },
+    "glbtNationalHotline": {
+      "name": "Línea Directa Nacional de Gays, Lesbianas, Bisexuales y Transgénero (GLBT)",
+      "description": "La línea directa nacional de gays, lesbianas, bisexuales y transgénero (GLBT) brinda apoyo telefónico, chat privado en línea y ayuda por correo electrónico. Te brindamos información objetiva y recursos locales en ciudades y pueblos de los Estados Unidos. Todos nuestros servicios son gratuitos y confidenciales. ¡Hablamos con personas que llaman de cualquier edad acerca de cualquier tema que surja como problemas en su relación, acoso en la escuela, problemas en su lugar de trabajo, ansiedad por el VIH / SIDA e información sobre sexo seguro, y mucho más! Nosotros tenemos un grupo diverso de voluntarios del grupo GLBT, operamos dos líneas directas nacionales, la Línea directa nacional de GLBT (1-888-843-4564) y la Línea de conversación nacional juvenil de GLBT (1-800-246-7743). Tambien usamos una sala de chat privada y voluntaria, que ayuda tanto a jóvenes y adultos. Nuestro correo electrónico es  help@GLBThotline.org"
```
Comment (jacobvenable) [inline packages/common/src/i18n/es.json:42]:
```suggestion
      "description": "La línea directa nacional de gays, lesbianas, bisexuales y transgénero (GLBT) brinda apoyo telefónico, chat privado en línea y ayuda por correo electrónico. Te brindamos información objetiva y recursos locales en ciudades y pueblos de los Estados Unidos. Todos nuestros servicios son gratuitos y confidenciales. ¡Hablamos con personas que llaman de cualquier edad acerca de cualquier tema que surja como problemas en su relación, acoso en la escuela, problemas en su lugar de trabajo, ansiedad por el VIH / SIDA e información sobre sexo seguro, y mucho más! Nosotros tenemos un grupo diverso de voluntarios del grupo GLBT, operamos dos líneas directas nacionales, la Línea directa nacional de GLBT (1-888-843-4564) y la Línea de conversación nacional juvenil de GLBT (1-800-246-7743). Tambien usamos una sala de chat privada y voluntaria, que ayuda tanto a jóvenes y adultos. Nuestro correo electrónico es help@GLBThotline.org"
```
Nit: Removing extra space.

File: packages/web/src/components/NotFound.tsx [hunk 3]
```diff
@@ -8,31 +9,36 @@ import SearchForm from "./SearchForm";
...
+          <SearchForm />
+        </Grid>
+        <Grid item>
+          <Typography>
+            <Trans
+              ns="notFound"
+              i18nKey="goHome"
+              components={{
+                MuiLink: <MuiLink component={RouterLink} to="/" />,
+              }}
```
Comment (jacobvenable) [inline packages/web/src/components/NotFound.tsx:35]: Huh, neat I hadn't used the `<Trans/>` component before 👍

Comment (jacobvenable) [reply packages/web/src/components/NotFound.tsx:35]: I am seeing that it doesn't re-render on language change though. What are your thoughts on this vs breaking this into separate keys instead with the `t` function instead?

**Review by jacobvenable — COMMENTED** [review]
Some small nits and one comment but this is looking good! 🎉

File: packages/web/src/components/NotFound.tsx (hunk 3 above)
Comment (paspheeris) [reply packages/web/src/components/NotFound.tsx:35]: That caught my eye as well and struck me as very odd, but from a bit more googling ([eg](https://github.com/i18next/react-i18next/issues/458)), it seems like that only applies if you're using i18next without HOCs or... something?
It shouldn't apply for us, and just to double check I tried toggling and it seems to rerender correctly!
i18nReRender

**Comment by paspheeris** [comment]
Thanks for the review, I think we should be good now!

**Review by jacobvenable — APPROVED** [review]
Awesome, thanks for the follow up!

**Pull request merged**